### ISSUE TITLE
Error in v1.1.0-beta.2 when accessing /v1/graphql: "resource does not exist"

### ISSUE THREAD

**Issue by barbalex** [hasura/graphql-engine#3734]
When using this docker-compose.yaml:
```yaml
version: '3.7'
services:
  graphql-engine:
    image: hasura/graphql-engine:v1.1.0-beta.2
    container_name: 'hasura'
    ports:
      - '8080:8080'
    restart: always
    environment:
      # database url to connect
      HASURA_GRAPHQL_DATABASE_URL: postgresql://user:secret@db-vermehrung-do-user-293456-0.db.ondigitalocean.com:25060/defaultdb?sslmode=require
      HASURA_GRAPHQL_ENABLE_CONSOLE: 'true' # set "false" to disable console
      HASURA_GRAPHQL_ADMIN_SECRET: secret
      HASURA_GRAPHQL_JWT_SECRET: '{"type": "RS512", "key": "-----BEGIN CERTIFICATE-----\ncertificate\n-----END CERTIFICATE-----\n"}'
    command:
      - graphql-engine
      - serve
  caddy:
    image: abiosoft/caddy:0.11.0
    container_name: 'caddy'
    depends_on:
      - 'graphql-engine'
    restart: always
    ports:
      - '80:80'
      - '443:443'
    volumes:
      - ./Caddyfile:/etc/Caddyfile
      - caddy_certs:/root/.caddy
volumes:
  caddy_certs:
```

i can access the console and that seems to work fine.
But I can't access `/v1/graphql` as the responce is:
```js
{
  "path": "$",
  "error": "resource does not exist",
  "code": "not-found"
}
```

**Comment by lexi-lambda** [comment]
I can’t reproduce the issue you describe. Could you provide more information?

**Comment by w-A-L-L-e** [comment]
The docker-compose.yml does indeed not work anymore. We had it set up on our project, then today all of sudden we're getting the error "code":"not-found". This happens only if you completely clear the database and start from scratch.

So I followed the recommended tutorial and it also happens when using the default docker-compose.yml.
You don't notice it until you do (as docker keeps around database volumes in its cache) a full clean:
```
$ docker system prune --volumes
WARNING! This will remove:
  - all stopped containers
  - all networks not used by at least one container
  - all volumes not used by at least one container
  - all dangling images
  - all dangling build cache

Are you sure you want to continue? [y/N] y
Deleted Containers:
f1782cb2dd96ccd8573877d7eac1c3c3404dce7ca51b05024ac70c2bf877c254
361c45d127c9876caec5e384c8f1f728508a7e7c2c5e8f45465501f89a7c05fd
78d60d55e4f52ec643a3389f1452f48f07b5648dc9ae508c76ebe42d8250f73a
eb6bddff747269e166ec65a4ded18f32f32d3b08bb6d03763d49c620efeaf1f4
c0e70dae8117c8bafb0e9ca407330c9f17eed9834d0e4a86f2265a7a30b29829

Deleted Networks:
sc-hasura_default

Deleted Volumes:
sc-hasura_redis
sc-hasura_postgres
sc-hasura_db_data
```
Now fetch 'official' the docker-compose
``` 
curl https://raw.githubusercontent.com/hasura/graphql-engine/stable/install-manifests/docker-compose/docker-compose.yaml -o docker-compose.yml
```



Then do a docker-compose up --build and watch it crash:

```
$ docker-compose up --build
Starting sc-hasura_postgres_1 ... done
Starting sc-hasura_graphql-engine_1 ... done
Attaching to sc-hasura_postgres_1, sc-hasura_graphql-engine_1
postgres_1        | 
postgres_1        | PostgreSQL Database directory appears to contain a database; Skipping initialization
postgres_1        | 
graphql-engine_1  | {"timestamp":"2020-09-09T17:31:30.000+0000","level":"info","type":"startup","detail":{"kind":"migration-apply","info":"database url for migrations is set by HASURA_GRAPHQL_DATABASE_URL"}}
postgres_1        | 2020-09-09 17:31:29.612 UTC [1] LOG:  starting PostgreSQL 12.4 (Debian 12.4-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
postgres_1        | 2020-09-09 17:31:29.612 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgres_1        | 2020-09-09 17:31:29.612 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres_1        | 2020-09-09 17:31:29.616 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres_1        | 2020-09-09 17:31:29.640 UTC [24] LOG:  database system was shut down at 2020-09-09 17:31:27 UTC
postgres_1        | 2020-09-09 17:31:29.649 UTC [1] LOG:  database system is ready to accept connections
graphql-engine_1  | {"timestamp":"2020-09-09T17:31:30.000+0000","level":"info","type":"startup","detail":{"kind":"migration-apply","info":"migrations server port env var is not set, defaulting to 9691"}}
graphql-engine_1  | {"timestamp":"2020-09-09T17:31:30.000+0000","level":"info","type":"startup","detail":{"kind":"migration-apply","info":"server timeout is not set, defaulting to 30 seconds"}}
graphql-engine_1  | {"timestamp":"2020-09-09T17:31:30.000+0000","level":"info","type":"startup","detail":{"kind":"migration-apply","info":"starting graphql engine temporarily on port 9691"}}
graphql-engine_1  | {"timestamp":"2020-09-09T17:31:30.000+0000","level":"info","type":"startup","detail":{"kind":"migration-apply","info":"waiting 30 for 9691 to be ready"}}
graphql-engine_1  | {"type":"startup","timestamp":"2020-09-09T17:31:30.444+0000","level":"info","detail":{"kind":"server_configuration","info":{"live_query_options":{"batch_size":100,"refetch_delay":1},"transaction_isolation":"ISOLATION LEVEL READ COMMITTED","plan_cache_options":{"plan_cache_size":null},"enabled_log_types":["http-log","websocket-log","startup","webhook-log","query-log"],"server_host":"HostAny","enable_allowlist":false,"log_level":"info","auth_hook_mode":"GET","use_prepared_statements":true,"unauth_role":null,"stringify_numeric_types":false,"enabled_apis":["metadata"],"enable_telemetry":false,"enable_console":true,"auth_hook":"http://localhost:8080/graphql_authorize","jwt_secret":null,"cors_config":{"allowed_origins":"*","disabled":false,"ws_read_cookie":null},"console_assets_dir":null,"admin_secret_set":true,"port":9691}}}
graphql-engine_1  | {"type":"startup","timestamp":"2020-09-09T17:31:30.444+0000","level":"info","detail":{"kind":"postgres_connection","info":{"retries":1,"database_url":"postgres://postgres:...@postgres:5432/postgres"}}}
graphql-engine_1  | {"timestamp":"2020-09-09T17:31:32.000+0000","level":"info","type":"startup","detail":{"kind":"migration-apply","info":"port 9691 is ready"}}
graphql-engine_1  | {"timestamp":"2020-09-09T17:31:32.000+0000","level":"info","type":"startup","detail":{"kind":"migration-apply","info":"checking for migrations directory"}}
graphql-engine_1  | {"timestamp":"2020-09-09T17:31:32.000+0000","level":"info","type":"startup","detail":{"kind":"migration-apply","info":"applying migrations from /avo-migrations"}}
graphql-engine_1  | {"type":"startup","timestamp":"2020-09-09T17:31:31.651+0000","level":"info","detail":{"kind":"db_migrate","info":"Successfully initialized the catalog (at version 36)."}}
graphql-engine_1  | {"type":"startup","timestamp":"2020-09-09T17:31:31.651+0000","level":"info","detail":{"kind":"schema-sync","info":{"thread_id":"ThreadId 17","instance_id":"18de7f21-f748-4bd7-8eca-ee3153e21731","message":"listener thread started"}}}
graphql-engine_1  | {"type":"startup","timestamp":"2020-09-09T17:31:31.651+0000","level":"info","detail":{"kind":"schema-sync","info":{"thread_id":"ThreadId 18","instance_id":"18de7f21-f748-4bd7-8eca-ee3153e21731","message":"processor thread started"}}}
graphql-engine_1  | {"type":"startup","timestamp":"2020-09-09T17:31:31.651+0000","level":"info","detail":{"kind":"event_triggers","info":"preparing data"}}
graphql-engine_1  | {"type":"startup","timestamp":"2020-09-09T17:31:31.651+0000","level":"info","detail":{"kind":"event_triggers","info":"starting workers"}}
graphql-engine_1  | {"type":"startup","timestamp":"2020-09-09T17:31:31.651+0000","level":"info","detail":{"kind":"scheduled_triggers","info":"preparing data"}}
graphql-engine_1  | {"type":"startup","timestamp":"2020-09-09T17:31:31.651+0000","level":"info","detail":{"kind":"server","info":{"time_taken":1.3057803,"message":"starting API server"}}}
graphql-engine_1  | {"level":"info","msg":"Help us improve Hasura! The cli collects anonymized usage stats which\nallow us to keep improving Hasura at warp speed. To opt-out or read more,\nvisit https://hasura.io/docs/1.0/graphql/manual/guides/telemetry.html\n","time":"2020-09-09T17:31:32Z"}
graphql-engine_1  | {"level":"info","msg":"Applying migrations...","time":"2020-09-09T17:31:32Z"}
graphql-engine_1  | {"type":"http-log","timestamp":"2020-09-09T17:31:31.651+0000","level":"error","detail":{"operation":{"error":{"path":"$","error":"resource does not exist","code":"not-found"},"request_id":"4da115b3-5db2-4b07-8bd6-1c848955793a","response_size":65,"raw_query":""},"http_info":{"status":404,"http_version":"HTTP/1.1","url":"/v1alpha1/config","ip":"127.0.0.1","method":"GET","content_encoding":null}}}
```

**Comment by w-A-L-L-e** [comment]
This version does seem to work. So we're reverting to this v 1.3.0-cli-migrations:
```
cat docker-compose.yml
version: '3.6'
services:
  postgres:
    image: postgres:12
    restart: always
    volumes:
    - db_data:/var/lib/postgresql/data
    environment:
      POSTGRES_PASSWORD: postgrespassword
  graphql-engine:
    #image: hasura/graphql-engine:v1.3.2
    image: hasura/graphql-engine:v1.3.0.cli-migrations
    ports:
    - "8080:8080"
    depends_on:
    - "postgres"
    restart: always
    environment:
      HASURA_GRAPHQL_DATABASE_URL: postgres://postgres:postgrespassword@postgres:5432/postgres
      ## enable the console served by server
      HASURA_GRAPHQL_ENABLE_CONSOLE: "true" # set to "false" to disable console
      ## enable debugging mode. It is recommended to disable this in production
      HASURA_GRAPHQL_DEV_MODE: "true"
      HASURA_GRAPHQL_ENABLED_LOG_TYPES: startup, http-log, webhook-log, websocket-log, query-log
      ## uncomment next line to set an admin secret
      # HASURA_GRAPHQL_ADMIN_SECRET: myadminsecretkey
volumes:
  db_data:

```

**Comment by tirumaraiselvan** [comment]
@w-A-L-L-e 

From the error:

```
graphql-engine_1  | {"level":"info","msg":"Applying migrations...","time":"2020-09-09T17:31:32Z"}
graphql-engine_1  | {"type":"http-log","timestamp":"2020-09-09T17:31:31.651+0000","level":"error","detail":{"operation":{"error":{"path":"$","error":"resource does not exist","code":"not-found"},"request_id":"4da115b3-5db2-4b07-8bd6-1c848955793a","response_size":65,"raw_query":""},"http_info":{"status":404,"http_version":"HTTP/1.1","url":"/v1alpha1/config","ip":"127.0.0.1","method":"GET","content_encoding":null}}}
```

It appears that `v1alpha1/config` is being requested while migrations are being run. The cli.migrations image works as follows:

1. Starts Hasura in metadata only mode.
2. Applies migrations from the mounted directories
3. Starts Hasura in normal mode.

Only after 3. is complete, will you be able to run any http requests (which are non-metadata like `v1alpha1/config`). Why is this request being made during migration application?

**Comment by w-A-L-L-e** [comment]
@tirumaraiselvan good question. our second migration replaces the metadata is it possible that internally hasura then needs to have that path available?

First migration is an sql file database_structure.up.sql, something like:
```
CREATE SCHEMA app;
CREATE SCHEMA migrate;
CREATE SCHEMA shared;
...
```

Second migration is a import_metadata.up.yml file which brings metadata back to how we need it to be for above sql tables:
```
- type: replace_metadata       
  args:                                        
    allowlist: []             
    functions: []                                            
    query_collections: []        
    remote_schemas: []                                         
    tables:                 
    - array_relationships: []                   
      delete_permissions: []     
      event_triggers: []                      
      insert_permissions: []
      object_relationships: []
      select_permissions: []     
      table:                               
...
```
This used to work fine with older versions.

**Comment by tirumaraiselvan** [comment]
TBH, I think it might be a benign error. Is the container restarting by itself?

**Comment by w-A-L-L-e** [comment]
No it doesn't. However the migrations are not run either it just waits in this state without going further. I found if I go into the container with /sh and then manually run the /bin/docker_entrypoint.sh it does migrate and start to run correctly. Maybe an example docker-compose.yml with cli-migrate image and some actual demo migration files that add a table and update the metadata would be handy for future ref. For us its not super critical as with our .gitlab-ci file we run the migrations ourselves and for local dev I just remember the manual step of running entrypoint the first time.

**Comment by tirumaraiselvan** [comment]
@w-A-L-L-e Pls feel free to open a new issue for this. I think the original issue has gone stale so I am closing this.
